apps/remote_control/lib/lexical/remote_control/search/store/state.ex

```diff
@@ -0,0 +1,154 @@
+defmodule Lexical.RemoteControl.Search.Store.State do
+  alias Lexical.Project
+  alias Lexical.RemoteControl.Search.Fuzzy
+  alias Lexical.RemoteControl.Search.Store.Ets
+  require Logger
+
+  @index_path Path.join("indexes", "source.index.ets")
+
+  defstruct [:project, :index_path, :create_index, :update_index, :loaded?, :fuzzy, :ets_table]
+
+  def new(%Project{} = project, create_index, update_index) do
+    %__MODULE__{
+      create_index: create_index,
+      project: project,
+      index_path: index_path(project),
+      loaded?: false,
+      update_index: update_index
+    }
+    |> ensure_index_directory_exists()
+  end
+
+  def drop(%__MODULE__{}) do
+    Ets.drop()
```

Comment:
Since the backend is now a behaviour, maybe it's worth referencing `Ets` only once and using a module attribute?

```elixir
@backend Lexical.RemoteControl.Search.Store.Ets

...

def drop(%__MODULE__{}) do
  @backend.drop()
end
```

This would make experimenting with alternative backends a bit easier.

Reply:
That's for the next PR. I'm going to make an mnesia backend and see how that goes.

Reply:
Sounds good!